pkg/updater/service/docker.go

```diff
@@ -0,0 +1,159 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build !windows
+
+// Package service provides a way to interact with os services
+package service
+
+import (
+	"bytes"
+	"encoding/json"
+	"os"
+	"os/exec"
+	"path"
+
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+type dockerDaemonConfig map[string]interface{}
+type dockerRuntimesConfig map[string]map[string]interface{}
+
+const (
+	tmpDockerDaemonPath = "/tmp/daemon.json.tmp"
+	dockerDaemonPath    = "/etc/docker/daemon.json"
+)
+
+// setDockerConfig sets up the docker daemon to use the APM injector
+// even if docker isn't installed, to prepare for if it is installed
+// later
+func (a *apmInjectorInstaller) setDockerConfig() error {
```

Comment:
this misses extensive local tests which you can run using temp dir.
Having local test will allow you to check all scenarios and different existing files format + faster iterations if later we break this

same for apm_inject file manipulations